set.mm

```diff
@@ -528868,6 +528934,404 @@ Set induction (or epsilon induction)
       VNAVMVACVBVGUJABVGUEUFUGUHUIVEVKUKULABVGIVPRUMUNDVFUOUPUQABIURUS $.
   $}
 
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Transitive closure of a class
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  $( Declare new constant $)
+  $c t++ $.
+
+  $( Declare the syntax for the transitive closure of a class. $)
+  cttrcl $a class t++ R $.
+
+  ${
+    $d R f n m x y $.
+    $( Define the transitive closure of a class.  This is the smallest
+       relationship containing ` R ` and having the transitive property.
+       Definition from [Levy] p. 59, who denotes it as ` R * ` and calls it the
+       "ancestral" of ` R ` .  (Contributed by Scott Fenton, 17-Oct-2024.) $)
+    df-ttrcl $a |- t++ R = { <. x , y >. | E. n e. ( _om \ 1o )
+     E. f ( f Fn suc n /\ ( ( f ` (/) ) = x /\ ( f ` n ) = y ) /\
+     A. m e. n ( f ` m ) R ( f ` suc m ) ) } $.
+  $}
+
+  ${
+    $d R f m n x y $.  $d S f m n x y $.
+    $( Equality theorem for transitive closure.  (Contributed by Scott Fenton,
+       17-Oct-2024.) $)
+    ttrcleq $p |- ( R = S -> t++ R = t++ S ) $=
+      ( vf vn vx vy vm wceq csuc wfn cfv wbr wral w3a wex copab cttrcl df-ttrcl
+      cv wrex com c1o cdif breq ralbidv 3anbi3d exbidv rexbidv opabbidv 3eqtr4g
+      c0 wa ) ABHZCSZDSZIJZUKUNKESHUOUNKFSHULZGSZUNKZURIUNKZALZGUOMZNZCOZDUAUBU
+      CZTZEFPUPUQUSUTBLZGUOMZNZCOZDVETZEFPAQBQUMVFVKEFUMVDVJDVEUMVCVICUMVBVHUPU
+      QUMVAVGGUOUSUTABUDUEUFUGUHUIEFACGDREFBCGDRUJ $.
+  $}
+
+  ${
+    $d R f m n y z $.  $d x f m n y z $.
+    nfttrcl.1 $e |- F/_ x R $.
+    $( Bound variable hypothesis builder for transitive closure.  (Contributed
+       by Scott Fenton, 17-Oct-2024.) $)
+    nfttrcl $p |- F/_ x t++ R $=
+      ( vf vn vy vz vm cttrcl cv csuc wfn c0 cfv wceq wa wbr wral nfcv nfv cdif
+      w3a wex com c1o wrex copab df-ttrcl nfbr nfralw nf3an nfrex nfopab nfcxfr
+      nfex ) ABIDJZEJZKLZMUPNFJOUQUPNGJOPZHJZUPNZUTKUPNZBQZHUQRZUBZDUCZEUDUEUAZ
+      UFZFGUGFGBDHEUHVHFGAVFAEVGAVGSVEADURUSVDAURATUSATVCAHUQAUQSAVAVBBAVASCAVB
+      SUIUJUKUOULUMUN $.
+  $}
+
+  ${
+    $d R f n m x y $.
+    $( The transitive closure of a class is a relationship.  (Contributed by
+       Scott Fenton, 17-Oct-2024.) $)
+    relttrcl $p |- Rel t++ R $=
+      ( vf vn vx vy vm cv csuc wfn c0 cfv wceq wa wbr wral w3a wex com c1o cdif
+      wrex cttrcl df-ttrcl relopabi ) BGZCGZHIJUEKDGLUFUEKEGLMFGZUEKUGHUEKANFUF
+      OPBQCRSTUADEAUBDEABFCUCUD $.
+  $}
+
+  ${
+    $d A x y n f a $.  $d B x y n f a $.  $d R x y n f a $.
+    $( Binary relationship form of transitive closure.  (Contributed by Scott
```

Comment:
```suggestion
    $( Binary relation form of transitive closure.  (Contributed by Scott
```
Or: "Characterization of elements of the transitive closure of a relation."

Reply:
Done